src/main/java/edu/hm/hafner/analysis/parser/GnuFortranParser.java

```diff
@@ -6,46 +6,125 @@
-            + "\\n" + "[^\\n]+\\n" // The offending line itself. NOCHECKSTYLE
-            + "[^\\n]+\\n" // The '1' and/or '2' corresponding to the column of the error locus.
-            + "(Warning|Error|Fatal Error|Internal Error at \\(1\\)):[\\s\\n]([^\\n]+)\\n";
+    private static final String MESSAGE_START_PATTERN = "(?s)^(.+\\.[^:]+):(\\d+)(?:\\.(\\d+)(?:-(\\d+))?)?:";
+    // file:file followed by Optional coulm and range followed by a colon.
+
+    /** Optional part of the category. */
     private static final Pattern LINE_PATTERN = Pattern.compile(" at \\(\\d\\)");
+    /** Include lines between message start and end. */
+    private static final Pattern INCLUDE_LINE_PATTERN = Pattern.compile("(?: {4}Included at .+)");
+    /** Simple regex to match any non empty lines which are required before the message end. */
+    private static final Pattern NON_EMPTY_LINE_PATTERN = Pattern.compile(".+");
+    /** Regex to match the category and the actual error message itself. */
+    private static final Pattern ERROR_MESSAGE_PATTERN = Pattern.compile("(Warning|Error|Fatal Error|Internal Error at \\(1\\)):\\s?(.*)");
 
     /**
      * Creates a new instance of {@link GnuFortranParser}.
      */
     public GnuFortranParser() {
-        super(GFORTRAN_MSG_PATTERN, true);
+        super(MESSAGE_START_PATTERN);
     }
 
+    /**
+     * This Function is called once the start of a possible error message is detected.
+     * It uses the provided lookaheadStream to check if the following lines match the message too.
+     * The lines from the lookaheadStream are only consumed if they are successfully matched to a part
+     * of the error message, if they can not be matched the function returns without creating an issue.
+     *
+     * @param matcher the regular expression matcher
+     * @param lookahead the lookahead stream to read additional lines
+     * @param builder the issue builder to use
+     *
+     * @return Issue if the rest of the message was sucesfully matched too
+     */
     @Override
-    protected Optional<Issue> createIssue(final Matcher matcher, final IssueBuilder builder) {
-        String category = LINE_PATTERN.matcher(matcher.group(5)).replaceAll("");
-        return builder.setFileName(matcher.group(1))
-                .setColumnStart(matcher.group(3))
-                .setColumnEnd(matcher.group(4))
-                .setLineStart(matcher.group(2))
+    protected Optional<Issue> createIssue(final Matcher matcher, final LookaheadStream lookahead, final IssueBuilder builder) {
+        // Gather location of the error.
+        String fileName = matcher.group(1);
+        String lineStart = matcher.group(2);
+        String columStart = matcher.group(3);
+        String columEnd = matcher.group(4);
+
+        if (!lookahead.hasNext()) {
+            // This does not match the whole message, return without issue.
+            return Optional.empty();
+        }
+        String currentLine = lookahead.peekNext();
```

Comment:
It is possible to simplify this by calling next() here as well as in line 80 and removing line 76 completely, isn't it?

Reply:
peekNext is used to avoid consuming lines which are not matched by the parser. If next is used and the parser determines that the line did not match the message currently being parsed the start of another message might be lost.
Using next() everywhere would make the code easier to read / better to understand, but could lead to some unexpected behaviour in edge casses.

Reply:
You can use `hasNext(String)` to simplify this